base/gmp.jl

```diff
@@ -521,8 +521,11 @@ hex(n::BigInt) = base(16, n)
 for f in (:bin, :oct, :dec, :hex)
     @eval function ($f)(n::BigInt, pad::Int)
         b = IOBuffer()
-        res = ($f)(n)
+        res = ($f)(abs(n))
```

Comment:
The `BigInt` type is not actually immutable as that would prevent attaching a finalizer to it. BigInts are only immutable by convention – they do not support mutating operations. Ultimately, we'd like to change that precisely so that we can do optimizations like that, but it's not currently possible.